pkg/networkservice/connectioncontextkernel/ethernetcontext/macaddress/server.go

```diff
@@ -57,26 +58,25 @@ type setVppMacServer struct{}
 //                              +---------------------------+
 //
 func NewServer() networkservice.NetworkServiceServer {
-	return &setVppMacServer{}
+	return &setKernelMacServer{}
 }
 
-func (s *setVppMacServer) Request(ctx context.Context, request *networkservice.NetworkServiceRequest) (*connection.Connection, error) {
-	conf := vppagent.Config(ctx)
-	conn, err := next.Client(ctx).Request(ctx, request)
-	if err != nil {
-		return nil, err
-	}
+func (s *setKernelMacServer) Request(ctx context.Context, request *networkservice.NetworkServiceRequest) (*connection.Connection, error) {
 	if mechanism := kernel.ToMechanism(request.GetConnection().GetMechanism()); mechanism != nil {
-		index := len(conf.GetLinuxConfig().GetInterfaces()) - 1
-		conf.GetLinuxConfig().GetInterfaces()[index+1].PhysAddress = conn.GetContext().GetEthernetContext().GetDstMac()
+		s.setDstMac(ctx, request.GetConnection())
 	}
-	return conn, nil
+	return next.Server(ctx).Request(ctx, request)
 }
 
-func (s *setVppMacServer) Close(ctx context.Context, conn *connection.Connection) (*empty.Empty, error) {
-	conf := vppagent.Config(ctx)
-	if mechanism := kernel.ToMechanism(conn.GetMechanism()); mechanism != nil && len(conf.GetLinuxConfig().GetInterfaces()) > 0 {
-		conf.GetLinuxConfig().GetInterfaces()[len(conf.GetVppConfig().GetInterfaces())-1].PhysAddress = conn.GetContext().GetEthernetContext().GetDstMac()
+func (s *setKernelMacServer) Close(ctx context.Context, conn *connection.Connection) (*empty.Empty, error) {
+	if mechanism := kernel.ToMechanism(conn.GetMechanism()); mechanism != nil {
+		s.setDstMac(ctx, conn)
 	}
-	return next.Client(ctx).Close(ctx, conn)
+	return next.Server(ctx).Close(ctx, conn)
+}
+
+func (s *setKernelMacServer) setDstMac(ctx context.Context, conn *connection.Connection) {
```

Comment:
Please leave this inline rather than pulling it out into a separate setDstMac function here.  It's small and simple and more comprehensible inline in Request/Close

Reply:
fixed.